File: push/src/main/java/com/xiaomi/smack/packet/IQ.java
package com.xiaomi.smack.packet;

import android.os.Bundle;
import com.xiaomi.push.service.PushConstants;
import com.xiaomi.smack.util.StringUtils;
import java.util.HashMap;
import java.util.Map;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/smack/packet/IQ.class */
public class IQ extends Packet {
    private final Map<String, String> attributes;
    private Type type;

    /* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/smack/packet/IQ$Type.class */
    public static class Type {
        private String value;
        public static final Type GET = new Type("get");
        public static final Type SET = new Type("set");
        public static final Type RESULT = new Type("result");
        public static final Type ERROR = new Type(Message.MSG_TYPE_ERROR);
        public static final Type COMMAND = new Type("command");

        private Type(String str) {
            this.value = str;
        }

        public static Type fromString(String str) {
            if (str == null) {
                return null;
            }
            String lowerCase = str.toLowerCase();
            Type type = GET;
            if (type.toString().equals(lowerCase)) {
                return type;
            }
            Type type2 = SET;
            if (type2.toString().equals(lowerCase)) {
                return type2;
            }
            Type type3 = ERROR;
            if (type3.toString().equals(lowerCase)) {
                return type3;
            }
            Type type4 = RESULT;
            if (type4.toString().equals(lowerCase)) {
                return type4;
            }
            Type type5 = COMMAND;
            if (type5.toString().equals(lowerCase)) {
                return type5;
            }
            return null;
        }

        public String toString() {
            return this.value;
        }
    }

    public IQ() {
        this.type = Type.GET;
        this.attributes = new HashMap();
    }

    public IQ(Bundle bundle) {
        super(bundle);
        this.type = Type.GET;
        this.attributes = new HashMap();
        if (bundle.containsKey(PushConstants.EXTRA_IQ_TYPE)) {
            this.type = Type.fromString(bundle.getString(PushConstants.EXTRA_IQ_TYPE));
        }
    }

    public static IQ createErrorResponse(IQ iq, XMPPError xMPPError) {
        if (iq.getType() != Type.GET && iq.getType() != Type.SET) {
            throw new IllegalArgumentException("IQ must be of type 'set' or 'get'. Original IQ: " + iq.toXML());
        }
        IQ iq2 = new IQ() { // from class: com.xiaomi.smack.packet.IQ.2
            @Override // com.xiaomi.smack.packet.IQ
            public String getChildElementXML() {
                return iq.getChildElementXML();
            }
        };
        iq2.setType(Type.ERROR);
        iq2.setPacketID(iq.getPacketID());
        iq2.setFrom(iq.getTo());
        iq2.setTo(iq.getFrom());
        iq2.setError(xMPPError);
        return iq2;
    }

    public static IQ createResultIQ(IQ iq) {
        if (iq.getType() != Type.GET && iq.getType() != Type.SET) {
            throw new IllegalArgumentException("IQ must be of type 'set' or 'get'. Original IQ: " + iq.toXML());
        }
        IQ iq2 = new IQ() { // from class: com.xiaomi.smack.packet.IQ.1
            @Override // com.xiaomi.smack.packet.IQ
            public String getChildElementXML() {
                return null;
            }
        };
        iq2.setType(Type.RESULT);
        iq2.setPacketID(iq.getPacketID());
        iq2.setFrom(iq.getTo());
        iq2.setTo(iq.getFrom());
        return iq2;
    }

    public String getAttribute(String str) {
        String str2;
        synchronized (this) {
            str2 = this.attributes.get(str);
        }
        return str2;
    }

    public String getChildElementXML() {
        return null;
    }

    public Type getType() {
        return this.type;
    }

    public void setAttribute(String str, String str2) {
        synchronized (this) {
            this.attributes.put(str, str2);
        }
    }

    public void setAttributes(Map<String, String> map) {
        synchronized (this) {
            this.attributes.putAll(map);
        }
    }

    public void setType(Type type) {
        if (type == null) {
            this.type = Type.GET;
        } else {
            this.type = type;
        }
    }

    @Override // com.xiaomi.smack.packet.Packet
    public Bundle toBundle() {
        Bundle bundle = super.toBundle();
        Type type = this.type;
        if (type != null) {
            bundle.putString(PushConstants.EXTRA_IQ_TYPE, type.toString());
        }
        return bundle;
    }

    @Override // com.xiaomi.smack.packet.Packet
    public String toXML() {
        StringBuilder sb = new StringBuilder();
        sb.append("<iq ");
        if (getPacketID() != null) {
            sb.append("id=\"" + getPacketID() + "\" ");
        }
        if (getTo() != null) {
            sb.append("to=\"");
            sb.append(StringUtils.escapeForXML(getTo()));
            sb.append("\" ");
        }
        if (getFrom() != null) {
            sb.append("from=\"");
            sb.append(StringUtils.escapeForXML(getFrom()));
            sb.append("\" ");
        }
        if (getChannelId() != null) {
            sb.append("chid=\"");
            sb.append(StringUtils.escapeForXML(getChannelId()));
            sb.append("\" ");
        }
        for (Map.Entry<String, String> entry : this.attributes.entrySet()) {
            sb.append(StringUtils.escapeForXML(entry.getKey()));
            sb.append("=\"");
            sb.append(StringUtils.escapeForXML(entry.getValue()));
            sb.append("\" ");
        }
        if (this.type == null) {
            sb.append("type=\"get\">");
        } else {
            sb.append("type=\"");
            sb.append(getType());
            sb.append("\">");
        }
        String childElementXML = getChildElementXML();
        if (childElementXML != null) {
            sb.append(childElementXML);
        }
        sb.append(getExtensionsXML());
        XMPPError error = getError();
        if (error != null) {
            sb.append(error.toXML());
        }
        sb.append("</iq>");
        return sb.toString();
    }
}

File: legacy/src/main/java/com/xiaomi/smack/packet/IQ.kt
package com.xiaomi.smack.packet

import android.os.Bundle
import com.xiaomi.push.service.PushConstants
import com.xiaomi.smack.util.StringUtils
import java.util.HashMap

/*
 * Current override reference: com.xiaomi.xmsf 0.3.17-20260410000745 (versionCode 1003003000),
 * base.apk sha256 f3d72b6f5e1427ceecd3147a051d58e4dc95bb528397d486658e01cad9f7e590,
 * JADX path: com.xiaomi.xmsf/current/base/sources/com/xiaomi/smack/packet/IQ.java
 * No stock 7.4.67-C same-path source was found in the split source tree.
 */
open class IQ : Packet {
    private val attributes: MutableMap<String, String> = HashMap()
    private var type: Type? = Type.GET

    class Type private constructor(private val value: String) {
        override fun toString(): String = value

        companion object {
            @JvmField val GET = Type("get")
            @JvmField val SET = Type("set")
            @JvmField val RESULT = Type("result")
            @JvmField val ERROR = Type(Message.MSG_TYPE_ERROR)
            @JvmField val COMMAND = Type("command")

            @JvmStatic
            fun fromString(str: String?): Type? {
                if (str == null) {
                    return null
                }
                val lowerCase = str.lowercase()
                return when {
                    GET.toString() == lowerCase -> GET
                    SET.toString() == lowerCase -> SET
                    ERROR.toString() == lowerCase -> ERROR
                    RESULT.toString() == lowerCase -> RESULT
                    COMMAND.toString() == lowerCase -> COMMAND
                    else -> null
                }
            }
        }
    }

    constructor() : super()

    constructor(bundle: Bundle?) : super(bundle) {
        if (bundle != null && bundle.containsKey(PushConstants.EXTRA_IQ_TYPE)) {
            type = Type.fromString(bundle.getString(PushConstants.EXTRA_IQ_TYPE))
        }
    }

    fun getAttribute(str: String): String? = synchronized(this) {
        attributes[str]
    }

    open fun getChildElementXML(): String? = null

    fun getType(): Type? = type

    fun setAttribute(str: String, str2: String) {
        synchronized(this) {
            attributes[str] = str2
        }
    }

    fun setAttributes(map: Map<String, String>) {
        synchronized(this) {
            attributes.putAll(map)
        }
    }

    fun setType(type: Type?) {
        this.type = type ?: Type.GET
    }

    override fun toBundle(): Bundle {
        val bundle = super.toBundle()
        type?.let { bundle.putString(PushConstants.EXTRA_IQ_TYPE, it.toString()) }
        return bundle
    }

    override fun toXML(): String {
        val sb = StringBuilder()
        sb.append("<iq ")
        if (packetID != null) {
            sb.append("id=\"").append(packetID).append("\" ")
        }
        if (to != null) {
            sb.append("to=\"")
            sb.append(StringUtils.escapeForXML(to))
            sb.append("\" ")
        }
        if (from != null) {
            sb.append("from=\"")
            sb.append(StringUtils.escapeForXML(from))
            sb.append("\" ")
        }
        if (channelId != null) {
            sb.append("chid=\"")
            sb.append(StringUtils.escapeForXML(channelId))
            sb.append("\" ")
        }
        for ((key, value) in attributes) {
            sb.append(StringUtils.escapeForXML(key))
            sb.append("=\"")
            sb.append(StringUtils.escapeForXML(value))
            sb.append("\" ")
        }
        if (type == null) {
            sb.append("type=\"get\">")
        } else {
            sb.append("type=\"")
            sb.append(getType())
            sb.append("\">")
        }
        val childElementXML = getChildElementXML()
        if (childElementXML != null) {
            sb.append(childElementXML)
        }
        sb.append(extensionsXML)
        val error = error
        if (error != null) {
            sb.append(error.toXML())
        }
        sb.append("</iq>")
        return sb.toString()
    }

    companion object {
        @JvmStatic
        fun createErrorResponse(iq: IQ, xMPPError: XMPPError): IQ {
            if (iq.getType() != Type.GET && iq.getType() != Type.SET) {
                throw IllegalArgumentException("IQ must be of type 'set' or 'get'. Original IQ: " + iq.toXML())
            }
            val iq2 = object : IQ() {
                override fun getChildElementXML(): String? {
                    return iq.getChildElementXML()
                }
            }
            iq2.setType(Type.ERROR)
            iq2.packetID = iq.packetID
            iq2.from = iq.to
            iq2.to = iq.from
            iq2.error = xMPPError
            return iq2
        }

        @JvmStatic
        fun createResultIQ(iq: IQ): IQ {
            if (iq.getType() != Type.GET && iq.getType() != Type.SET) {
                throw IllegalArgumentException("IQ must be of type 'set' or 'get'. Original IQ: " + iq.toXML())
            }
            val iq2 = object : IQ() {
                override fun getChildElementXML(): String? = null
            }
            iq2.setType(Type.RESULT)
            iq2.packetID = iq.packetID
            iq2.from = iq.to
            iq2.to = iq.from
            return iq2
        }
    }
}
